app/javascript/components/student/editor/ExercismMonacoEditor.tsx

```diff
@@ -88,6 +87,27 @@ export function ExercismMonacoEditor({
         connection.onClose(() => disposable.dispose())
       },
     })
+  }, [document, listen])
+
+  useEffect(() => {
+    window.addEventListener('load', startLanguageClient)
+
+    return () => {
+      window.removeEventListener('load', startLanguageClient)
```

Comment:
Also problematic with this approach: it doesn't "disconnect" on "unload". Probably is what you wanted, but that's not what's happening.